src/components/tools/Map/VectorLayer.svelte

```diff
@@ -52,10 +62,16 @@
 
   if (enableClick) {
     map.on("click", layer.id, function (e) {
-      dispatch("overlayclick", e.features[0].id);
-      map.flyTo({
-        center: e.features[0].geometry.coordinates,
-      });
+      const feature = e.features && e.feature.length ? e.feature[0] : null;
+      const center = getFeatureCenter(feature);
+      try {
```

Comment:
I wouldn't wrap the `try catch` around `dispatch` and the `map.flyTo`. I think the more brittle place is where the geometry is being checked or parsed in `getFeatureCenter`